src/common/snippets/src/lowered/pass/define_buffer_clusters.cpp

```diff
@@ -353,6 +355,15 @@ bool DefineBufferClusters::run(lowered::LinearIR& linear_ir, lowered::LinearIR::
         }
     }
 
+    for (size_t cluster_id = 0; cluster_id < m_clusters.size(); ++cluster_id) {
+        const auto& cluster = m_clusters[cluster_id];
+        std::for_each(cluster.cbegin(), cluster.cend(), [&cluster_id](const ExpressionPtr& buffer_expr) {
+            const auto& buffer = ov::as_type_ptr<op::Buffer>(buffer_expr->get_node());
```

Comment:
Just a question: do we have a reason for storing `Expression` in buffer cluster? I see that we always have to call `get_node()` and cast to `Buffer` -- maybe we can avoid it by storing `Buffer` nodes in `BufferCluster`?

Reply:
Good idea, thank you. I thought about that but we have to work sometimes with port connectors of expressions from clusters.
https://github.com/openvinotoolkit/openvino/blob/009c7fcf0fab42a4085bad4f7e5b2faec82be40a/src/common/snippets/src/lowered/pass/define_buffer_clusters.cpp#L305-L307

So it seems like we have to contain expressions in cluster.